models/migrations/v67.go

```diff
@@ -0,0 +1,13 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+package migrations
+
+import (
+	"github.com/go-xorm/xorm"
+)
+
+func cleanUnlinkedWebhookAndHookTasks(x *xorm.Engine) error {
+	_, err := x.Exec(`DELETE FROM webhook WHERE (SELECT id FROM repository WHERE id = webhook.repo_id) ISNULL;DELETE FROM hook_task WHERE (SELECT id FROM repository WHERE id = hook_task.repo_id) ISNULL;`)
```

Comment:
I don't think this will work in PostgreSQL.

SQL should be something like:
```
DELETE FROM webhook WHERE repo_id not in (SELECT id FROM repository);
DELETE FROM hook_task WHERE repo_id not in (SELECT id FROM repository);
```